scripts/nmt/train_transformer.py

```diff
@@ -163,7 +163,7 @@ def load_cached_dataset(prefix):
     cached_file_path = os.path.join(_C.CACHE_PATH, prefix + '.npz')
     if os.path.exists(cached_file_path):
         print('Load cached data from {}'.format(cached_file_path))
-        dat = np.load(cached_file_path)
+        dat = np.load(cached_file_path, encoding='bytes')
```

Comment:
What if Py3 was used to save the file? Wouldn't encoding=bytes be wrong (ie. the correct encoding in that case would be utf-8)? 
I believe for other parts of the library we choose to default to 'utf-8' even if the user is running Py2. So instead of changing the load part maybe we should prevent writing something in bytes encoding?

Reply:
`np.load` does not support `encoding='utf-8'`, the default encoding is ascii.

Reply:
the cached dataset is stored in int32 format, so I think byte is ok

Reply:
What is stored in the numpy archive? Are these all strings? If so it may be better to save as np.array(X, dtype='O') and make sure that these are unicode strings in Py2. Otherwise all strings are padded to same length

Reply:
Arrays of index of word in vocabulary.

Reply:
But why do you need to specify an encoding then? Encoding seems only useful when loading Python 2 generated pickled files in Python 3. Ie. it should not be necessary if your dtype is int32?